### DISCUSSION TITLE
Error Running Example Code with Generic Python REPL

### DISCUSSION THREAD

**Discussion by drpjm** [py5coding/py5generator#337] [Q&A] [answered]
I just ran through the install process on my Mac. I run Ventura with Homebrew and all of the pre-reqs are installed. When I run the [quick example](https://py5coding.org/content/install.html#quick-example) in a virtual environment and repl, I received the following error from py5:
```python
Unable to obtain source code for setup(). Either make it obtainable or create a settings() function for calls to size(), fullscreen(), etc.
py5 encountered an error in your code:

File "<stdin>", line 2, in setup

RuntimeError: Cannot call the size() method here. Either move it to a settings() function or move it to closer to the start of setup().
```
I did not see anything specific to this issue in the [Mac OSX notes](https://py5coding.org/content/osx_users.html#generic-python-interpreter). Is there some other thing I need to do to run this from the python repl?

**Comment by hx2A** [comment]
Welcome, @drpjm !

I think you are running the example code in the generic Python interpreter. Can you use IPython instead, or put the example code in a file and run it from the command line with `python test.py`?

py5, like Processing, does some magic to allow you to put method calls such as `size()` in the `setup()` function instead of `settings()`, where they really belong. The magic will do some surgery on your `setup()` function to create a `settings()` method for you and a new `setup()` function. For this magic to work, py5 needs to have access to the source code of `setup()`, and to do that, it needs the source code to be written to a file. IPython does some stuff behind the scenes to meet this need, which is why the `setup()` stuff works just fine when run with IPyython. The generic Python interpreter does not do this. It can also provide less useful stack traces.

I looked at the installation documentation and I see the "Here is a quick py5 example to test that everything works" statement should say to test with IPython or by saving the code to a file. I will update this so others don't also run into the same problem.

**Reply by hx2A** [reply]
https://github.com/py5coding/py5book/issues/99

**Reply by drpjm** [reply]
Thank you for the added details! And thanks also for taking the time to update the docs. As I read them, it seemed that it would work fine in a standard REPL, but I will try as a script instead. I will also attempt the IPython approach

**Reply by hx2A** [reply, marked as answer]
Great, I expect that will work well for you.

I just updated the install docs so future users will benefit from this discussion. Thanks again for your interest in py5 and your contribution!
